### PR TITLE
Add pip metadata for scallopy

### PR DESCRIPTION
**Pull request by ControlNet**
The current installation of `scallopy` requires cloning the repository then use `make` or `maturin` to install the library.

In this PR, I added the `pyproject.toml` to `scallopy` subfolder, and added a declaration of used features in `core/src/lib.rs`, allowing it can be directly installed by `pip`. 

With these changes, users can install `scallopy` by
```
pip install ./etc/scallopy
```

Or even **NO NEED TO CLONE**.
```
pip install "git+https://github.com/scallop-lang/scallop.git#egg=scallopy&subdirectory=etc/scallopy"
```

I believe this contribution can improve the package distribution process. As it is compiling from source, there should be no Python version and CPU architecture constrain.

### CODE REVIEW

File: core/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #![feature(extract_if)]
 #![feature(hash_extract_if)]
 #![feature(proc_macro_span)]
+#![feature(iter_repeat_n)]
```
Comment (Liby99) [inline core/src/lib.rs:5]: I wonder why is this needed because the latest Rust distribution seems to have stabilized the feature.

**Review by Liby99 — COMMENTED** [review]
Thank you so much for the PR and the change to scallopy looks good to me. In terms of this specific feature flag can you double check why is it needed? Other than that I'm happy to merge the PR.

File: core/src/lib.rs (hunk 1 above)
Comment (ControlNet) [reply core/src/lib.rs:5]: Hi, without adding it, the installing scallopy using pip will fail. Sorry I'm not very familiar with rust. I think you can also try running `pip install ./etc/scallopy` and see what happened if not adding this feature.

Comment (Liby99) [reply core/src/lib.rs:5]: Thanks. The checks have passed anyway so I'll move ahead and merge the PR.

**Review by Liby99 — APPROVED** [review]
LGTM!
